### PR TITLE
Loading indicators

### PR DESCRIPTION
**Pull request by jonathanmcdaniel**
Adds in Ladda buttons for form submissions so that the user is given feedback when api calls are happening in the background. Also moved boostrap into a new /vendor directory so that we can keep non-node related 3rd party packages in a standard location. /vendor will hold ladda stuff, bootstrap, and fontawesome. I also changed fontawesome to use a local version instead of the cdn. The cdn seemed to be having issues and having it locally will give us more control over it anyway :).

**NOTE: the notifications pull request will have to go before this one, and I will likely have a merge conflict on the package.json that I will have to resolve before this one is pushed.**

### CODE REVIEW

File: Frontend/app/user/registration/registration.component.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export class RegistrationComponent {
   mentiiConfig = new MentiiConfig();
   submitInProgress = false;
   regSuccess = false;
+  isLoading = false;
```
Comment (asp78) [inline Frontend/app/user/registration/registration.component.ts]: Is there a difference between 'isLoading' and 'submitInProgress'? They have awfully similar names

Comment (jonathanmcdaniel) [inline Frontend/app/user/registration/registration.component.ts]: uh the other one was already there. Let me look and Ill get rid of one of them

File: Frontend/vendor/fontawesome/css/font-awesome.css [hunk 2]
```diff
@@ -0,0 +1,2337 @@
+/*!
```
Comment (asp78) [inline Frontend/vendor/fontawesome/css/font-awesome.css]: Why are we storing this if we already have the min.css?

File: Frontend/app/user/registration/registration.component.ts (hunk 1 above)
Comment (jonathanmcdaniel) [inline Frontend/app/user/registration/registration.component.ts]: updated

File: Frontend/vendor/fontawesome/css/font-awesome.css (hunk 2 above)
Comment (jonathanmcdaniel) [inline Frontend/vendor/fontawesome/css/font-awesome.css]: removed

File: Frontend/app/user/registration/registration.component.ts [hunk 3]
```diff
@@ -12,8 +12,8 @@ import { UserService } from '../user.service';
 export class RegistrationComponent {
   model = new RegistrationModel('', '', '');
   mentiiConfig = new MentiiConfig();
-  submitInProgress = false;
```
Comment (aaronmsegal) [inline Frontend/app/user/registration/registration.component.ts]: I think submitInProgress is more descriptive then isLoading.

**Pull request merged**